astroquery/mast/core.py

```diff
@@ -1304,7 +1339,69 @@ def _download_files(self, products, base_dir, cache=True):
             url = None
 
             try:
-                self._download_file(dataUrl, localPath, cache=cache)
+                s3_fetch_ok = False
+                if boto3 is not None and dataUri.startswith("mast:HST/product"):
+                    # The following is a mishmash of BaseQuery._download_file and s3 access through boto
+
+                    try:
+                        obs_id = obs_id.lower()
+
+                        # magic associations logic per Brian - 0-9/a-e we convert to 0
+                        magicValues = "123456789abcde"
+                        if obs_id[-1] in magicValues:
+                            print("WARNING: IPPPSSOOT Magic in %s" % dataUri)
```

Comment:
use `warnings.warn` instead of `print("WARNING")`

Reply:
Is this really a warning though?  Perhaps `log.info` would be appropriate?

Reply:
In this case the user has found a data product who's filename has probably been mangled along the way.  This is fragile and I think we should let the users know that there is a chance that this download won't work through S3 due to design issues at ST.

Reply:
OK, understood.   I still recommend `log.warning` or `warnings.warn`

Reply:
To keep with the rest of the module I would suggest warnings.warn, and maybe build a new warning "OutputWarning" or something.